In [1]:
import pandas as pd


Importing Dataset

In [ ]:
df = pd.read_excel("/content/Data - Survey Monkey Output.xlsx",sheet_name="Edited_Data")
df

Created a copy of our dataset incase we need to revert back

In [3]:
df_modified = df.copy()
df_modified

Respondent ID          Start Date            End Date  Email Address  \
0       5379192392 2021-01-22 12:01:17 2021-01-22 12:40:34            NaN   
1       2658722536 2021-01-22 06:56:37 2021-01-22 07:34:10            NaN   
2       4044163394 2021-01-22 06:35:18 2021-01-22 06:47:32            NaN   
3       5535865599 2021-01-21 21:29:32 2021-01-21 21:40:24            NaN   
4       3356802928 2021-01-21 17:26:39 2021-01-21 17:44:40            NaN   
..             ...                 ...                 ...            ...   
193     7940065082 2021-01-11 06:19:18 2021-01-11 06:32:59            NaN   
194     5157705612 2021-01-11 06:19:14 2021-01-11 06:33:29            NaN   
195     9920755555 2021-01-11 06:18:48 2021-01-11 06:27:27            NaN   
196     6638341389 2021-01-11 06:19:01 2021-01-11 06:33:06            NaN   
197     8114622230 2021-01-11 06:18:50 2021-01-11 06:26:46            NaN   

     First Name  Last Name  Custom Data 1  \
0           NaN        NaN            NaN   
1           NaN        NaN            NaN   
2           NaN        NaN            NaN   
3           NaN        NaN            NaN   
4           NaN        NaN            NaN   
..          ...        ...            ...   
193         NaN        NaN            NaN   
194         NaN        NaN            NaN   
195         NaN        NaN            NaN   
196         NaN        NaN            NaN   
197         NaN        NaN            NaN   

    Identify which division you work in. - Response  \
0                                    Infrastructure   
1                                           Finance   
2                                    Infrastructure   
3                                    Infrastructure   
4                                   Port Operations   
..                                              ...   
193                                  Infrastructure   
194                                         Finance   
195                                 Port Operations   
196                                  Infrastructure   
197                          Information Technology   

    Identify which division you work in. - Other (please specify)  \
0                                                  NaN              
1                                                  NaN              
2                                                  NaN              
3                                                  NaN              
4                                                  NaN              
..                                                 ...              
193                                                NaN              
194                                                NaN              
195                                                NaN              
196                                                NaN              
197                                                NaN              

    Which of the following best describes your position level? - Response  \
0                                                Staff                      
1                                                Staff                      
2                                      Department Lead                      
3                                              Manager                      
4                                              Manager                      
..                                                 ...                      
193                                    Department Lead                      
194                                              Staff                      
195                                              Staff                      
196                                            Manager                      
197                                              Staff                      

     ... Question 29 - Response 8 Question 29 - Response 9  \
0    ...                      NaN                 Answer 8   


In [4]:
df_modified.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 198 entries, 0 to 197
Data columns (total 100 columns):
 #   Column                                                                         Non-Null Count  Dtype         
---  ------                                                                         --------------  -----         
 0   Respondent ID                                                                  198 non-null    int64         
 1   Start Date                                                                     198 non-null    datetime64[ns]
 2   End Date                                                                       198 non-null    datetime64[ns]
 3   Email Address                                                                  0 non-null      float64       
 4   First Name                                                                     0 non-null      float64       
 5   Last Name                                                                      0 non

In [5]:
df_modified.columns

Index(['Respondent ID', 'Start Date', 'End Date', 'Email Address',
       'First Name', 'Last Name', 'Custom Data 1',
       'Identify which division you work in. - Response',
       'Identify which division you work in. - Other (please specify)',
       'Which of the following best describes your position level? - Response',
       'Which generation are you apart of? - Response',
       'Please select the gender in which you identify. - Response',
       'Which duration range best aligns with your tenure at your company? - Response',
       'Which of the following best describes your employment type? - Response',
       'Question 1 - Response', 'Question 2 - Response',
       'Question 3 - Open-Ended Response', 'Question 4 - Response',
       'Question 4 - Other (please specify)', 'Question 5 - Response 1',
       'Question 5 - Response 2', 'Question 5 - Response 3',
       'Question 5 - Response 4', 'Question 5 - Response 5',
       'Question 5 - Response 6', 'Question 6 - Response 1

Dropping columns we dont need

In [6]:
columns_to_drop = ['Start Date', 'End Date', 'Email Address','First Name', 'Last Name', 'Custom Data 1']
columns_to_drop

['Start Date',
 'End Date',
 'Email Address',
 'First Name',
 'Last Name',
 'Custom Data 1']

In [7]:
df_modified.drop(columns = columns_to_drop,inplace = True)

In [8]:
df_modified

Respondent ID Identify which division you work in. - Response  \
0       5379192392                                  Infrastructure   
1       2658722536                                         Finance   
2       4044163394                                  Infrastructure   
3       5535865599                                  Infrastructure   
4       3356802928                                 Port Operations   
..             ...                                             ...   
193     7940065082                                  Infrastructure   
194     5157705612                                         Finance   
195     9920755555                                 Port Operations   
196     6638341389                                  Infrastructure   
197     8114622230                          Information Technology   

    Identify which division you work in. - Other (please specify)  \
0                                                  NaN              
1                                                  NaN              
2                                                  NaN              
3                                                  NaN              
4                                                  NaN              
..                                                 ...              
193                                                NaN              
194                                                NaN              
195                                                NaN              
196                                                NaN              
197                                                NaN              

    Which of the following best describes your position level? - Response  \
0                                                Staff                      
1                                                Staff                      
2                                      Department Lead                      
3                                              Manager                      
4                                              Manager                      
..                                                 ...                      
193                                    Department Lead                      
194                                              Staff                      
195                                              Staff                      
196                                            Manager                      
197                                              Staff                      

    Which generation are you apart of? - Response  \
0           Generation X (born between 1965-1980)   
1                                             NaN   
2           Generation X (born between 1965-1980)   
3             Millennial (born between 1981-2000)   
4           Generation X (born between 1965-1980)   
..                                            ...   
193          Baby Boomer (born between 1946-1964)   
194           Millennial (born between 1981-2000)   
195           Millennial (born between 1981-2000)   
196           Millennial (born between 1981-2000)   
197                          Prefer not to answer   

    Please select the gender in which you identify. - Response  \
0                                                 Male           
1                                                  NaN           
2                                                 Male           
3                                           Non-Binary           
4                                               Female           
..                                                 ...           
193                                               Male           
194                                             Female           
195                                             Female           
196                                             Female           
197                                            

Unpivoting our dataframe


In [9]:
id_vars = list(df_modified.columns)[0 : 8]
id_vars

['Respondent ID',
 'Identify which division you work in. - Response',
 'Identify which division you work in. - Other (please specify)',
 'Which of the following best describes your position level? - Response',
 'Which generation are you apart of? - Response',
 'Please select the gender in which you identify. - Response',
 'Which duration range best aligns with your tenure at your company? - Response',
 'Which of the following best describes your employment type? - Response']

In [10]:
value_vars = list(df_modified.columns)[8 : ]
#value_vars

In [11]:
df_melted = pd.melt(df_modified,id_vars=id_vars,value_vars = value_vars,var_name ="Question + Subquestion" ,value_name = "Answer")
df_melted

Respondent ID Identify which division you work in. - Response  \
0         5379192392                                  Infrastructure   
1         2658722536                                         Finance   
2         4044163394                                  Infrastructure   
3         5535865599                                  Infrastructure   
4         3356802928                                 Port Operations   
...              ...                                             ...   
17023     7940065082                                  Infrastructure   
17024     5157705612                                         Finance   
17025     9920755555                                 Port Operations   
17026     6638341389                                  Infrastructure   
17027     8114622230                          Information Technology   

      Identify which division you work in. - Other (please specify)  \
0                                                    NaN              
1                                                    NaN              
2                                                    NaN              
3                                                    NaN              
4                                                    NaN              
...                                                  ...              
17023                                                NaN              
17024                                                NaN              
17025                                                NaN              
17026                                                NaN              
17027                                                NaN              

      Which of the following best describes your position level? - Response  \
0                                                  Staff                      
1                                                  Staff                      
2                                        Department Lead                      
3                                                Manager                      
4                                                Manager                      
...                                                  ...                      
17023                                    Department Lead                      
17024                                              Staff                      
17025                                              Staff                      
17026                                            Manager                      
17027                                              Staff                      

      Which generation are you apart of? - Response  \
0             Generation X (born between 1965-1980)   
1                                               NaN   
2             Generation X (born between 1965-1980)   
3               Millennial (born between 1981-2000)   
4             Generation X (born between 1965-1980)   
...                                             ...   
17023          Baby Boomer (born between 1946-1964)   
17024           Millennial (born between 1981-2000)   
17025           Millennial (born between 1981-2000)   
17026           Millennial (born between 1981-2000)   
17027                          Prefer not to answer   

      Please select the gender in which you identify. - Response  \
0                                                   Male           
1                                                    NaN           
2                                                   Male           
3                                             Non-Binary           
4                                                 Female           
...                                                  ...           
17023                                               Male           
17024                                             Female           
17025                                             Female           

In [12]:
df_melted.columns

Index(['Respondent ID', 'Identify which division you work in. - Response',
       'Identify which division you work in. - Other (please specify)',
       'Which of the following best describes your position level? - Response',
       'Which generation are you apart of? - Response',
       'Please select the gender in which you identify. - Response',
       'Which duration range best aligns with your tenure at your company? - Response',
       'Which of the following best describes your employment type? - Response',
       'Question + Subquestion', 'Answer'],
      dtype='object')

In [ ]:
df_melted_2 = df_melted.copy()
df_melted_2

Created a Seperate column which only contains Questions

In [17]:
df_melted_2[["Question", "to_drop"]] = df_melted_2["Question + Subquestion"].str.split(" - ", expand=True)

In [19]:
df_melted_2.drop(columns = "to_drop",inplace = True)
df_melted_2

Respondent ID Identify which division you work in. - Response  \
0         5379192392                                  Infrastructure   
1         2658722536                                         Finance   
2         4044163394                                  Infrastructure   
3         5535865599                                  Infrastructure   
4         3356802928                                 Port Operations   
...              ...                                             ...   
17023     7940065082                                  Infrastructure   
17024     5157705612                                         Finance   
17025     9920755555                                 Port Operations   
17026     6638341389                                  Infrastructure   
17027     8114622230                          Information Technology   

      Identify which division you work in. - Other (please specify)  \
0                                                    NaN              
1                                                    NaN              
2                                                    NaN              
3                                                    NaN              
4                                                    NaN              
...                                                  ...              
17023                                                NaN              
17024                                                NaN              
17025                                                NaN              
17026                                                NaN              
17027                                                NaN              

      Which of the following best describes your position level? - Response  \
0                                                  Staff                      
1                                                  Staff                      
2                                        Department Lead                      
3                                                Manager                      
4                                                Manager                      
...                                                  ...                      
17023                                    Department Lead                      
17024                                              Staff                      
17025                                              Staff                      
17026                                            Manager                      
17027                                              Staff                      

      Which generation are you apart of? - Response  \
0             Generation X (born between 1965-1980)   
1                                               NaN   
2             Generation X (born between 1965-1980)   
3               Millennial (born between 1981-2000)   
4             Generation X (born between 1965-1980)   
...                                             ...   
17023          Baby Boomer (born between 1946-1964)   
17024           Millennial (born between 1981-2000)   
17025           Millennial (born between 1981-2000)   
17026           Millennial (born between 1981-2000)   
17027                          Prefer not to answer   

      Please select the gender in which you identify. - Response  \
0                                                   Male           
1                                                    NaN           
2                                                   Male           
3                                             Non-Binary           
4                                                 Female           
...                                                  ...           
17023                                               Male           
17024                                             Female           
17025                                             Female           

Finding number of people responded for a question

In [31]:
respondents = df_melted_2[df_melted_2["Answer"].notna()]
respondents = respondents.groupby("Question")["Respondent ID"].nunique().reset_index()


In [32]:
respondents.rename(columns={"Respondent ID":"Respondents"},inplace = True)
respondents

Question  Respondents
0    Question 1          119
1   Question 10          198
2   Question 11          164
3   Question 12          114
4   Question 13          108
5   Question 14          105
6   Question 15          114
7   Question 16          117
8   Question 17          135
9   Question 18          109
10  Question 19          157
11   Question 2          118
12  Question 20          105
13  Question 21          127
14  Question 22          160
15  Question 23          120
16  Question 24          195
17  Question 25          198
18  Question 26          193
19  Question 27          153
20  Question 28          119
21  Question 29          198
22   Question 3          117
23  Question 30          182
24   Question 4          161
25   Question 5          194
26   Question 6          196
27   Question 7          162
28   Question 8          190
29   Question 9          188

In [33]:
df_merged = pd.merge(df_melted_2,respondents,how="left" ,left_on="Question" , right_on="Question" ,)
print("Before",len(df_melted_2))
print("After",len(df_merged))
df_merged

Before 17028
After 17028


Respondent ID Identify which division you work in. - Response  \
0         5379192392                                  Infrastructure   
1         2658722536                                         Finance   
2         4044163394                                  Infrastructure   
3         5535865599                                  Infrastructure   
4         3356802928                                 Port Operations   
...              ...                                             ...   
17023     7940065082                                  Infrastructure   
17024     5157705612                                         Finance   
17025     9920755555                                 Port Operations   
17026     6638341389                                  Infrastructure   
17027     8114622230                          Information Technology   

      Identify which division you work in. - Other (please specify)  \
0                                                    NaN              
1                                                    NaN              
2                                                    NaN              
3                                                    NaN              
4                                                    NaN              
...                                                  ...              
17023                                                NaN              
17024                                                NaN              
17025                                                NaN              
17026                                                NaN              
17027                                                NaN              

      Which of the following best describes your position level? - Response  \
0                                                  Staff                      
1                                                  Staff                      
2                                        Department Lead                      
3                                                Manager                      
4                                                Manager                      
...                                                  ...                      
17023                                    Department Lead                      
17024                                              Staff                      
17025                                              Staff                      
17026                                            Manager                      
17027                                              Staff                      

      Which generation are you apart of? - Response  \
0             Generation X (born between 1965-1980)   
1                                               NaN   
2             Generation X (born between 1965-1980)   
3               Millennial (born between 1981-2000)   
4             Generation X (born between 1965-1980)   
...                                             ...   
17023          Baby Boomer (born between 1946-1964)   
17024           Millennial (born between 1981-2000)   
17025           Millennial (born between 1981-2000)   
17026           Millennial (born between 1981-2000)   
17027                          Prefer not to answer   

      Please select the gender in which you identify. - Response  \
0                                                   Male           
1                                                    NaN           
2                                                   Male           
3                                             Non-Binary           
4                                                 Female           
...                                                  ...           
17023                                               Male           
17024                                             Female           
17025                                             Female           

Finding Number of people gave same answer per question

In [34]:
same_answer = df_melted_2 # [df_melted_2["Answer"].notna()]
same_answer = same_answer.groupby(["Question + Subquestion","Answer"])["Respondent ID"].nunique().reset_index()
same_answer.rename(columns={"Respondent ID":"Same Answer"},inplace = True)
same_answer

Question + Subquestion    Answer  Same Answer
0      Question 1 - Response  Answer 1           14
1      Question 1 - Response  Answer 2           10
2      Question 1 - Response  Answer 3           13
3      Question 1 - Response  Answer 4           17
4      Question 1 - Response  Answer 5           22
..                       ...       ...          ...
683  Question 9 - Response 4  Answer 4           16
684  Question 9 - Response 4  Answer 5           13
685  Question 9 - Response 4  Answer 6           14
686  Question 9 - Response 4  Answer 7           12
687  Question 9 - Response 4  Answer 8           19

[688 rows x 3 columns]

In [36]:
df_merged_2 = pd.merge(df_merged,same_answer,how="left" ,left_on=["Question + Subquestion","Answer"] , right_on=["Question + Subquestion","Answer"] ,)
print("Before",len(df_merged))
print("After",len(df_merged_2))
df_merged_2

Before 17028
After 17028


Respondent ID Identify which division you work in. - Response  \
0         5379192392                                  Infrastructure   
1         2658722536                                         Finance   
2         4044163394                                  Infrastructure   
3         5535865599                                  Infrastructure   
4         3356802928                                 Port Operations   
...              ...                                             ...   
17023     7940065082                                  Infrastructure   
17024     5157705612                                         Finance   
17025     9920755555                                 Port Operations   
17026     6638341389                                  Infrastructure   
17027     8114622230                          Information Technology   

      Identify which division you work in. - Other (please specify)  \
0                                                    NaN              
1                                                    NaN              
2                                                    NaN              
3                                                    NaN              
4                                                    NaN              
...                                                  ...              
17023                                                NaN              
17024                                                NaN              
17025                                                NaN              
17026                                                NaN              
17027                                                NaN              

      Which of the following best describes your position level? - Response  \
0                                                  Staff                      
1                                                  Staff                      
2                                        Department Lead                      
3                                                Manager                      
4                                                Manager                      
...                                                  ...                      
17023                                    Department Lead                      
17024                                              Staff                      
17025                                              Staff                      
17026                                            Manager                      
17027                                              Staff                      

      Which generation are you apart of? - Response  \
0             Generation X (born between 1965-1980)   
1                                               NaN   
2             Generation X (born between 1965-1980)   
3               Millennial (born between 1981-2000)   
4             Generation X (born between 1965-1980)   
...                                             ...   
17023          Baby Boomer (born between 1946-1964)   
17024           Millennial (born between 1981-2000)   
17025           Millennial (born between 1981-2000)   
17026           Millennial (born between 1981-2000)   
17027                          Prefer not to answer   

      Please select the gender in which you identify. - Response  \
0                                                   Male           
1                                                    NaN           
2                                                   Male           
3                                             Non-Binary           
4                                                 Female           
...                                                  ...           
17023                                               Male           
17024                                             Female           
17025                                             Female           

Filling Nan with 0 in Same Answer Cloumn

In [41]:
df_merged_2["Same Answer"].fillna(0,inplace = True)
df_merged_2["Answer"].fillna("Not Answered",inplace = True)
df_merged_2

Respondent ID Identify which division you work in. - Response  \
0         5379192392                                  Infrastructure   
1         2658722536                                         Finance   
2         4044163394                                  Infrastructure   
3         5535865599                                  Infrastructure   
4         3356802928                                 Port Operations   
...              ...                                             ...   
17023     7940065082                                  Infrastructure   
17024     5157705612                                         Finance   
17025     9920755555                                 Port Operations   
17026     6638341389                                  Infrastructure   
17027     8114622230                          Information Technology   

      Identify which division you work in. - Other (please specify)  \
0                                                    NaN              
1                                                    NaN              
2                                                    NaN              
3                                                    NaN              
4                                                    NaN              
...                                                  ...              
17023                                                NaN              
17024                                                NaN              
17025                                                NaN              
17026                                                NaN              
17027                                                NaN              

      Which of the following best describes your position level? - Response  \
0                                                  Staff                      
1                                                  Staff                      
2                                        Department Lead                      
3                                                Manager                      
4                                                Manager                      
...                                                  ...                      
17023                                    Department Lead                      
17024                                              Staff                      
17025                                              Staff                      
17026                                            Manager                      
17027                                              Staff                      

      Which generation are you apart of? - Response  \
0             Generation X (born between 1965-1980)   
1                                               NaN   
2             Generation X (born between 1965-1980)   
3               Millennial (born between 1981-2000)   
4             Generation X (born between 1965-1980)   
...                                             ...   
17023          Baby Boomer (born between 1946-1964)   
17024           Millennial (born between 1981-2000)   
17025           Millennial (born between 1981-2000)   
17026           Millennial (born between 1981-2000)   
17027                          Prefer not to answer   

      Please select the gender in which you identify. - Response  \
0                                                   Male           
1                                                    NaN           
2                                                   Male           
3                                             Non-Binary           
4                                                 Female           
...                                                  ...           
17023                                               Male           
17024                                             Female           
17025                                             Female           

In [40]:
df_merged_2.columns

Index(['Respondent ID', 'Identify which division you work in. - Response',
       'Identify which division you work in. - Other (please specify)',
       'Which of the following best describes your position level? - Response',
       'Which generation are you apart of? - Response',
       'Please select the gender in which you identify. - Response',
       'Which duration range best aligns with your tenure at your company? - Response',
       'Which of the following best describes your employment type? - Response',
       'Question + Subquestion', 'Answer', 'Question', 'Respondents',
       'Same Answer'],
      dtype='object')

Renaming column Headers

In [42]:
df_cleaned_1 = df_merged_2.copy()
df_cleaned_1.rename(columns = {'Identify which division you work in. - Response':"Division Primary",
       'Identify which division you work in. - Other (please specify)':"Division Secondary",
       'Which of the following best describes your position level? - Response':"Position",
       'Which generation are you apart of? - Response':"Generation",
       'Please select the gender in which you identify. - Response':"Gender",
       'Which duration range best aligns with your tenure at your company? - Response':"Tenure",
       'Which of the following best describes your employment type? - Response':"Employment Type"},inplace = True)
df_cleaned_1

Respondent ID        Division Primary Division Secondary  \
0         5379192392          Infrastructure                NaN   
1         2658722536                 Finance                NaN   
2         4044163394          Infrastructure                NaN   
3         5535865599          Infrastructure                NaN   
4         3356802928         Port Operations                NaN   
...              ...                     ...                ...   
17023     7940065082          Infrastructure                NaN   
17024     5157705612                 Finance                NaN   
17025     9920755555         Port Operations                NaN   
17026     6638341389          Infrastructure                NaN   
17027     8114622230  Information Technology                NaN   

              Position                             Generation      Gender  \
0                Staff  Generation X (born between 1965-1980)        Male   
1                Staff                                    NaN         NaN   
2      Department Lead  Generation X (born between 1965-1980)        Male   
3              Manager    Millennial (born between 1981-2000)  Non-Binary   
4              Manager  Generation X (born between 1965-1980)      Female   
...                ...                                    ...         ...   
17023  Department Lead   Baby Boomer (born between 1946-1964)        Male   
17024            Staff    Millennial (born between 1981-2000)      Female   
17025            Staff    Millennial (born between 1981-2000)      Female   
17026          Manager    Millennial (born between 1981-2000)      Female   
17027            Staff                   Prefer not to answer        Male   

           Tenure     Employment Type    Question + Subquestion        Answer  \
0       0-2 years  Full time Employee     Question 1 - Response  Not Answered   
1       10+ years  Full time Employee     Question 1 - Response      Answer 4   
2       3-5 years  Full time Employee     Question 1 - Response      Answer 5   
3      5-10 years  Full time Employee     Question 1 - Response      Answer 1   
4       10+ years  Full time Employee     Question 1 - Response  Not Answered   
...           ...                 ...                       ...           ...   
17023   10+ years  Full time Employee  Question 30 - Response 3      Answer 8   
17024  5-10 years  Full time Employee  Question 30 - Response 3      Answer 6   
17025   3-5 years  Full time Employee  Question 30 - Response 3  Not Answered   
17026   3-5 years  Full time Employee  Question 30 - Response 3  Not Answered   
17027  5-10 years  Full time Employee  Question 30 - Response 3  Not Answered   

          Question  Respondents  Same Answer  
0       Question 1          119          0.0  
1       Question 1          119         17.0  
2       Question 1          119         22.0  
3       Question 1          119         14.0  
4       Question 1          119          0.0  
...            ...          ...          ...  
17023  Question 30          182         14.0  
17024  Question 30          182         20.0  
17025  Question 30          182          0.0  
17026  Question 30          182          0.0  
17027  Question 30          182          0.0  

[17028 rows x 13 columns]

In [44]:
df_cleaned_1.columns

Index(['Respondent ID', 'Division Primary', 'Division Secondary', 'Position',
       'Generation', 'Gender', 'Tenure', 'Employment Type',
       'Question + Subquestion', 'Answer', 'Question', 'Respondents',
       'Same Answer'],
      dtype='object')

Combining Divsion primary and Division Secondary Columns

In [64]:
df_cleaned_2 = df_cleaned_1.copy()
df_cleaned_2['Division Primary'] = df_cleaned_2.apply(lambda row: row['Division Secondary'] if pd.notna(row['Division Primary']) and row['Division Primary'] == 'Other (please specify)' else row['Division Primary'], axis=1)
df_cleaned_2

Respondent ID        Division Primary Division Secondary  \
0         5379192392          Infrastructure                NaN   
1         2658722536                 Finance                NaN   
2         4044163394          Infrastructure                NaN   
3         5535865599          Infrastructure                NaN   
4         3356802928         Port Operations                NaN   
...              ...                     ...                ...   
17023     7940065082          Infrastructure                NaN   
17024     5157705612                 Finance                NaN   
17025     9920755555         Port Operations                NaN   
17026     6638341389          Infrastructure                NaN   
17027     8114622230  Information Technology                NaN   

              Position                             Generation      Gender  \
0                Staff  Generation X (born between 1965-1980)        Male   
1                Staff                                    NaN         NaN   
2      Department Lead  Generation X (born between 1965-1980)        Male   
3              Manager    Millennial (born between 1981-2000)  Non-Binary   
4              Manager  Generation X (born between 1965-1980)      Female   
...                ...                                    ...         ...   
17023  Department Lead   Baby Boomer (born between 1946-1964)        Male   
17024            Staff    Millennial (born between 1981-2000)      Female   
17025            Staff    Millennial (born between 1981-2000)      Female   
17026          Manager    Millennial (born between 1981-2000)      Female   
17027            Staff                   Prefer not to answer        Male   

           Tenure     Employment Type    Question + Subquestion        Answer  \
0       0-2 years  Full time Employee     Question 1 - Response  Not Answered   
1       10+ years  Full time Employee     Question 1 - Response      Answer 4   
2       3-5 years  Full time Employee     Question 1 - Response      Answer 5   
3      5-10 years  Full time Employee     Question 1 - Response      Answer 1   
4       10+ years  Full time Employee     Question 1 - Response  Not Answered   
...           ...                 ...                       ...           ...   
17023   10+ years  Full time Employee  Question 30 - Response 3      Answer 8   
17024  5-10 years  Full time Employee  Question 30 - Response 3      Answer 6   
17025   3-5 years  Full time Employee  Question 30 - Response 3  Not Answered   
17026   3-5 years  Full time Employee  Question 30 - Response 3  Not Answered   
17027  5-10 years  Full time Employee  Question 30 - Response 3  Not Answered   

          Question  Respondents  Same Answer  
0       Question 1          119          0.0  
1       Question 1          119         17.0  
2       Question 1          119         22.0  
3       Question 1          119         14.0  
4       Question 1          119          0.0  
...            ...          ...          ...  
17023  Question 30          182         14.0  
17024  Question 30          182         20.0  
17025  Question 30          182          0.0  
17026  Question 30          182          0.0  
17027  Question 30          182          0.0  

[17028 rows x 13 columns]

In [65]:
df_cleaned_2[df_cleaned_2["Division Secondary"].notna()]["Division Secondary"]

13        Construction 
98       Rather not say
211       Construction 
296      Rather not say
409       Construction 
              ...      
16532    Rather not say
16645     Construction 
16730    Rather not say
16843     Construction 
16928    Rather not say
Name: Division Secondary, Length: 172, dtype: object

Cross checked,  
two people ansered as "other(specify)" = 2  
total quetion+subqutions = 86   
2*86 = 172

In [68]:
df_cleaned_2.drop(columns= "Division Secondary",inplace = True)
df_cleaned_2

Respondent ID        Division Primary         Position  \
0         5379192392          Infrastructure            Staff   
1         2658722536                 Finance            Staff   
2         4044163394          Infrastructure  Department Lead   
3         5535865599          Infrastructure          Manager   
4         3356802928         Port Operations          Manager   
...              ...                     ...              ...   
17023     7940065082          Infrastructure  Department Lead   
17024     5157705612                 Finance            Staff   
17025     9920755555         Port Operations            Staff   
17026     6638341389          Infrastructure          Manager   
17027     8114622230  Information Technology            Staff   

                                  Generation      Gender      Tenure  \
0      Generation X (born between 1965-1980)        Male   0-2 years   
1                                        NaN         NaN   10+ years   
2      Generation X (born between 1965-1980)        Male   3-5 years   
3        Millennial (born between 1981-2000)  Non-Binary  5-10 years   
4      Generation X (born between 1965-1980)      Female   10+ years   
...                                      ...         ...         ...   
17023   Baby Boomer (born between 1946-1964)        Male   10+ years   
17024    Millennial (born between 1981-2000)      Female  5-10 years   
17025    Millennial (born between 1981-2000)      Female   3-5 years   
17026    Millennial (born between 1981-2000)      Female   3-5 years   
17027                   Prefer not to answer        Male  5-10 years   

          Employment Type    Question + Subquestion        Answer  \
0      Full time Employee     Question 1 - Response  Not Answered   
1      Full time Employee     Question 1 - Response      Answer 4   
2      Full time Employee     Question 1 - Response      Answer 5   
3      Full time Employee     Question 1 - Response      Answer 1   
4      Full time Employee     Question 1 - Response  Not Answered   
...                   ...                       ...           ...   
17023  Full time Employee  Question 30 - Response 3      Answer 8   
17024  Full time Employee  Question 30 - Response 3      Answer 6   
17025  Full time Employee  Question 30 - Response 3  Not Answered   
17026  Full time Employee  Question 30 - Response 3  Not Answered   
17027  Full time Employee  Question 30 - Response 3  Not Answered   

          Question  Respondents  Same Answer  
0       Question 1          119          0.0  
1       Question 1          119         17.0  
2       Question 1          119         22.0  
3       Question 1          119         14.0  
4       Question 1          119          0.0  
...            ...          ...          ...  
17023  Question 30          182         14.0  
17024  Question 30          182         20.0  
17025  Question 30          182          0.0  
17026  Question 30          182          0.0  
17027  Question 30          182          0.0  

[17028 rows x 12 columns]

In [70]:
df_cleaned_2.iloc[16532]

Respondent ID                                        6225804357
Division Primary                                 Rather not say
Position                                                Manager
Generation                Generation X (born between 1965-1980)
Gender                                     Prefer not to answer
Tenure                                                10+ years
Employment Type                              Full time Employee
Question + Subquestion                 Question 30 - Response 1
Answer                                                 Answer 5
Question                                            Question 30
Respondents                                                 182
Same Answer                                                19.0
Name: 16532, dtype: object

Exporting to excel file

In [72]:
df_cleaned_2.to_excel("/content/Final_output_SurveyMonkey.xlsx",index = False)